src/mam/mod_mam.erl

```diff
@@ -588,18 +588,18 @@ lookup_messages_without_policy_violation_check(Host, #{search_text := SearchText
         true -> %% Use of disabled full text search
             {error, 'not-supported'};
         false ->
-            StartT = os:timestamp(),
+            StartT = erlang:monotonic_time(microsecond),
             R = mongoose_hooks:mam_lookup_messages(Host, {ok, {0, 0, []}}, Params),
-            Diff = timer:now_diff(os:timestamp(), StartT),
+            Diff = erlang:monotonic_time(microsecond) - StartT,
             mongoose_metrics:update(Host, [backends, ?MODULE, lookup], Diff),
             R
     end.
 
 -spec archive_message(jid:server(), mod_mam:archive_message_params()) -> ok | {error, timeout}.
 archive_message(Host, Params) ->
-    StartT = os:timestamp(),
+    StartT = erlang:monotonic_time(microsecond),
```

Comment:
Why switch from non-monotonic to monotonic? It should be consistent with the one used when archiving the message.

Reply:
Because this is to measure a time difference, non-monotonic doesn't ensure that they won't be the same value and it will report zero, even if it was actually not-zero.

Reply:
I understand, the natural choice here might be `erlang:timestamp`... it is still not clear for me when to use them and the current choice looks a bit random. I think some rules explaining them are needed before we change everything to make it consistent.

Reply:
Not really:
https://erlang.org/doc/man/erlang.html#timestamp-0 says `system_time/0,1` has better performance, and in the do and don't section of https://erlang.org/doc/apps/erts/time_correction.html#new-time-api it says `timestamp()` is there only to assume backwards compatibility with the code that uses the old format. We don't ned to create a triplet of bignums when we can have a single bignum consuming less memory and avoid converting from-and-to, if in the end what is stored in the database or send to metrics are bignums-as-timestamps.